[RHIDP-7712]: Add configuration files reference to the Configuring title (#2257)
redhat-developer/red-hat-developers-documentation-rhdh: modules/configure_configuring-rhdh/con-rhdh-configuration-files-overview.adoc

:_mod-docs-content-type: CONCEPT

[id="rhdh-configuration-files-overview_{context}"]
= {product} configuration files overview

[role="_abstract"]

> When configuring {product}, a common question is "Where do I configure this?" {product} uses several configuration files and Kubernetes resources, each serving a different purpose and provisioned differently depending on your deployment method.

Reviewer: ```suggestion
When configuring {product}, a common question is "Where do I configure this?". {product} uses several configuration files and Kubernetes resources, each serving a different purpose and provisioned differently depending on your deployment method.
```

> When configuring {product}, a common question is "Where do I configure this?". {product} uses several configuration files and Kubernetes resources, each serving a different purpose and provisioned differently depending on your deployment method.

Reviewer 2: Overview of the configuration files and Kubernetes resources used to manage {product} instances across different deployment methods.

Author: ```suggestion
Overview of the configuration files and Kubernetes resources used to manage {product} instances across different deployment methods.
```

{product-short} configuration files fall into two categories based on how they are consumed:

> Cluster-provisioned files:: Provisioned as Kubernetes resources (config maps or secrets) on the target cluster, typically by platform engineers. These files control the runtime behavior and deployment of your {product-very-short} instance. While the source files can be stored in a Git repository, they must be applied to the cluster to take effect.

Reviewer 3: ```suggestion
Cluster-provisioned files:: Provisioned as Kubernetes resources, such as config maps or secrets, on the target cluster, typically by platform engineers. These files control the runtime behavior and deployment of your {product-very-short} instance. While the source files can be stored in a Git repository, they must be applied to the cluster to take effect.
```

> Git-stored files:: Authored by developers and committed to source repositories. These files define catalog entities, software templates, and documentation configuration that {product-very-short} consumes at runtime.

Reviewer: Not sure I get this distinction.. Files like app-config, Backstage CR or Helm values files can also be stored in Git repositories, no?

Author: Can vs must. OK need to clarify.

> == Cluster-provisioned files

Reviewer 3: (Should this be created as a separate reference module?)

The following table lists configuration files that platform engineers create and provision as Kubernetes resources on the target cluster.

[cols="1,2,1,2,2", options="header", subs="+attributes"]
|===
| File
| Purpose
| Format
| Operator provisioning
| Helm chart provisioning

| `{my-app-config-file}`

> | Main application configuration (URLs, auth, catalog, database, plugin settings).

Reviewer 3: ```suggestion
| Main application configuration, including URLs, auth, catalog, database, plugin settings.
```

| YAML

> a| Config map referenced via `spec.application.appConfig.configMaps` in the `{product-custom-resource-type}` custom resource.

Reviewer 3: ```suggestion
a| Config map referenced in the `spec.application.appConfig.configMaps` field in the `{product-custom-resource-type}` custom resource.
```

> a| Inline via `upstream.backstage.appConfig` values, or external config map via `upstream.backstage.extraAppConfig`.

Reviewer 3: ```suggestion
a| Inline through `upstream.backstage.appConfig` values, or external config map through `upstream.backstage.extraAppConfig`.
```

> | Secrets (for example, `my-rhdh-secrets`)

Reviewer 3: ```suggestion
| Secrets, such as, `my-rhdh-secrets`)
```

| Authentication credentials, backend secret, database passwords, and other sensitive values.
| Kubernetes Secret
a| Secret referenced via `spec.application.extraEnvs.secrets` in the `{product-custom-resource-type}` CR.

> a| Referenced via `upstream.backstage.extraEnvs.secrets`.

Reviewer: For Helm, secrets can be injected as env vars using `upstream.backstage.extraEnvVarsSecrets` or even via `upstream.backstage.extraEnvVars` using a `secretKeyRef`.
See:
- https://github.com/redhat-developer/rhdh-chart/blob/b0e125c711a098b51a5263ab9b4b002d0bc16f95/charts/backstage/vendor/backstage/charts/backstage/values.yaml#L185-L186
- https://github.com/redhat-developer/rhdh-chart/blob/b0e125c711a098b51a5263ab9b4b002d0bc16f95/charts/backstage/vendor/backstage/charts/backstage/values.yaml#L179-L180
- https://kubernetes.io/docs/concepts/configuration/secret/#using-secrets-as-environment-variables

> a| Secret referenced via `spec.application.extraEnvs.secrets` in the `{product-custom-resource-type}` custom resource.

Reviewer 3: ```suggestion
a| Secret referenced in the `spec.application.extraEnvs.secrets` field in the `{product-custom-resource-type}` custom resource.
```

> a| Referenced via `upstream.backstage.extraEnvVarsSecrets` or `upstream.backstage.extraEnvVars` (using a `secretKeyRef`).

Reviewer 3: ```suggestion
a| Referenced through `upstream.backstage.extraEnvVarsSecrets` or `upstream.backstage.extraEnvVars` using a `secretKeyRef`.
```

| `dynamic-plugins.yaml`
| Enable, disable, and configure dynamic plugins.
| YAML

> a| Config map referenced via `spec.application.dynamicPluginsConfigMapName` in the `{product-custom-resource-type}` custom resource.

Reviewer 3: ```suggestion
a| Config map referenced in the `spec.application.dynamicPluginsConfigMapName` field in the `{product-custom-resource-type}` custom resource.
```

> a| Inline via `global.dynamic.plugins` Helm values.

Reviewer 3: ```suggestion
a| Inline through `global.dynamic.plugins` Helm values.
```

| `rbac-policy.csv`
| RBAC permission policies and role assignments (Casbin format).
| CSV
a| Config map mounted as file, path configured in `{my-app-config-file}` under `permission.rbac.policies-csv-file`.
a| Same.

| `rbac-conditional-policies.yaml`
| RBAC conditional policies with criteria-based filtering.
| YAML
a| Config map mounted as file, path configured in `{my-app-config-file}` under `permission.rbac.conditionalPoliciesFile`.
a| Same.

| `{product-custom-resource-type}` custom resource

> | Operator-side deployment configuration (references config files, routes, database, replicas).

Reviewer 3: ```suggestion
| Operator-side deployment configuration that references config files, routes, database, replicas.
```

> | YAML (Kubernetes custom resource)

Reviewer 3: ```suggestion
| YAML Kubernetes custom resource
```

a| Applied directly as a `{product-custom-resource-type}` custom resource.
| N/A

| Helm `values.yaml`

> | Helm-side deployment configuration (image, replicas, resources, routes).

Reviewer 3: ```suggestion
| Helm-side deployment configuration, including image, replicas, resources, routes.
```

| YAML
| N/A

> a| Passed via `helm install -f values.yaml` or `--set`.

Reviewer 3: ```suggestion
a| Passed by using `helm install -f values.yaml` or the `--set` option.
```

|===

== Application-consumed files

The following table lists configuration files that developers author and store in Git repositories. {product-very-short} reads these files directly from Git at runtime.

[cols="1,2,1,1,2", options="header", subs="+attributes"]
|===
| File
| Purpose
| Format
| Stored in
| Consumed by

| `catalog-info.yaml`
| Entity descriptor that defines components, APIs, systems, and their metadata and annotations.
| YAML
| Root of a source repository.

> | Software Catalog (through catalog locations or discovery providers).

Reviewer 3: ```suggestion
| Software Catalog through catalog locations or discovery providers.
```
